### PR TITLE
core: tests: Isolate Railjson generator

### LINKED ISSUE #3188: Isolate infra generator
### Description and goal

Make the infra generator a standalone library to be used by core tests but also research projects.

### Acceptance criteria

The library must be used by core tests and also usable as a standalone library for research projects

### PR DESCRIPTION
**Pull request by bloussou**
This PR is:
- moving raijson-generator in its own package under the `python/` folder
- linting it with flake8, isort and black
- making `railjson-genrator` a dependency of integration tests
- making `core` use the moved library to generate its examples
- updating script that loads small_infra and fast_rolling_stock

close #3188

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -133,10 +133,10 @@ test {
 }
 
 def genExamples = tasks.register("genExamples", Exec) {
-    inputs.files(fileTree('examples/generated/')).withPropertyName('sourceFiles')
+    workingDir '../python/railjson_generator/railjson_generator/'
+    inputs.files(fileTree('../python/railjson_generator/')).withPropertyName('sourceFiles')
```
Comment (eckter) [inline core/build.gradle]: It feels weird to have relative paths here. Maybe we could add a way to generate infras without specifying the full path of the input scripts?

**Review by eckter — DISMISSED** [review]

File: .github/workflows/core.yml [hunk 2]
```diff
@@ -25,7 +25,21 @@ jobs:
...
+        uses: actions/cache@v3
+        id: cached-poetry-railjson-dependencies
+        with:
+          path: python/railjson_generator/.venv
+          key: cached-poetry-railjson-dependencies-${{ hashFiles('python/railjson_generator/poetry.lock') }}
+      - name: Install dependencies
+        if: steps.cached-poetry-railjson-dependencies.outputs.cache-hit != 'true'
+        run: |
+          cd python/railjson_generator
+          poetry install
```
Comment (flomonster) [inline .github/workflows/core.yml]: Cache dependency can be handled with `actions/setup-python`.

File: README.md [hunk 3]
```diff
@@ -36,10 +36,11 @@ To compile and run the application with an example infrastructure:
...
 # generate and load an example infrastructure
-python3 -m venv .venv
-source .venv/bin/activate
-pip install geojson-pydantic
-scripts/generate-infra.sh small_infra
+cd python/railjson_generator/
+poetry install
+poetry shell
+cd ../../
+./scripts/generate-infra.sh small_infra
```
Comment (flomonster) [inline README.md]:
```suggestion
poetry --directory=python/railjson_generator install
poetry --directory=python/railjson_generator shell
./scripts/generate-infra.sh small_infra
```

File: python/railjson_generator/railjson_generator/schema/simulation/stop.py [hunk 4]
```diff
@@ -14,8 +14,7 @@ def __init__(self, duration, location=None, position=None):
         self.duration = duration
         if (location is None) == (position is None):
             raise RuntimeError(
-                "Stop instantiation: only one of the attributes 'location' and"
-                "'position' must be defined."
+                "Stop instantiation: only one of the attributes 'location' and" "'position' must be defined."
```
Comment (flomonster) [inline python/railjson_generator/railjson_generator/schema/simulation/stop.py]:
```suggestion
                "Stop instantiation: only one of the attributes 'location' and 'position' must be defined."
```

**Review by flomonster — COMMENTED** [review]
Awesome! I have minor suggestions.

File: README.md (hunk 3 above)
Comment (bloussou) [reply README.md]: nice :star:

**Review by flomonster — APPROVED** [review]
LGTM

**Review by eckter — APPROVED** [review]

**Pull request merged**